lib/App.js

```diff
@@ -109,22 +133,26 @@ class App {
 
     let promise    = Q();
     const response = new Response();
-    _.each(this._middleware, (route) => {
-      promise = promise.then(() => {
-        if (!response.ended) {
-          return route.execute({ request, response });
-        }
-      });
-    });
-    _.each(this._errorMiddleware, (route) => {
-      if (!route.isMatch(request)) {
-        return;
+    _.each(this._middleware, (registeredRoute) => {
+      const route = registeredRoute.route;
+      const errorMiddleware = registeredRoute.errorMiddleware;
+      if (!errorMiddleware) {
```

Comment:
Is this possible? The else block has a separate check where `if (!route.isMatch(request)) {
        return;
          }` and then error is being passed in if it is a catch. I moving the isMatch code in front of the then causes about 10 test failures. Is there something I am missing?